internal/comm/httptransport.go

```diff
@@ -0,0 +1,189 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package comm
+
+import (
+	"fmt"
+	"github.com/IBM-Blockchain/bcdb-server/config"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/logger"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/pkg/errors"
+	"go.etcd.io/etcd/etcdserver/api/rafthttp"
+	stats "go.etcd.io/etcd/etcdserver/api/v2stats"
+	etcd_types "go.etcd.io/etcd/pkg/types"
+	"go.etcd.io/etcd/raft/raftpb"
+	"net"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+//go:generate counterfeiter -o mocks/consensus_listener.go --fake-name ConsensusListener . ConsensusListener
+
+type ConsensusListener interface {
+	rafthttp.Raft
+}
+
+type HTTPTransport struct {
+	localConf *config.LocalConfiguration
+
+	mutex             sync.Mutex
+	consensusListener ConsensusListener
+	clusterConfig     *types.ClusterConfig
+
+	raftID uint64
+
+	transport  *rafthttp.Transport
+	httpServer *http.Server
+
+	stopCh chan struct{} // signals HTTPTransport to shutdown
+	doneCh chan struct{} // signals HTTPTransport shutdown complete
+
+	logger *logger.SugarLogger
+}
+
+type Config struct {
+	LocalConf *config.LocalConfiguration
+	Logger    *logger.SugarLogger
+}
+
+func NewHTTPTransport(config *Config) *HTTPTransport {
+	if config.LocalConf.Replication.TLS.Enabled {
+		config.Logger.Panic("TLS not supported yet")
+	}
+
+	tr := &HTTPTransport{
+		logger:    config.Logger,
+		localConf: config.LocalConf,
+		stopCh:    make(chan struct{}),
+		doneCh:    make(chan struct{}),
+	}
+	//TODO
+	return tr
+}
+
+func (p *HTTPTransport) SetConsensusListener(l ConsensusListener) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	if p.consensusListener != nil {
+		return errors.New("ConsensusListener already set")
+	}
+	p.consensusListener = l
+
+	return nil
+}
+
+//TODO dynamic re-config
+func (p *HTTPTransport) UpdateClusterConfig(clusterConfig *types.ClusterConfig) error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	if p.clusterConfig != nil {
+		return errors.New("dynamic re-config of http transport is not supported yet")
+	}
+
+	p.clusterConfig = clusterConfig
+	return nil
+}
+
+func (p *HTTPTransport) Start() error {
+	p.mutex.Lock()
+	defer p.mutex.Unlock()
+
+	if p.consensusListener == nil {
+		p.logger.Panic("Must set ConsensusListener before Start()")
+	}
+
+	if p.clusterConfig == nil {
+		p.logger.Panic("Must update ClusterConfig before Start()")
+	}
+
+	//TODO export to a config util
+	var foundRaftID bool
+	for _, member := range p.clusterConfig.ConsensusConfig.Members {
+		if member.NodeId == p.localConf.Server.Identity.ID {
+			p.raftID = member.RaftId
+			foundRaftID = true
+			break
+		}
+	}
```

Comment:
Can the member be an observer and want to start the HTTP server to receive the blocks? 

I think that the observer part is not necessary and adds unnecessary complexity.